cms/djangoapps/contentstore/views/certificates.py

```diff
@@ -0,0 +1,380 @@
+                signatory['id'] = generate_int_id(used_ids=used_ids)
+            used_ids.append(signatory['id'])
+
+        return certificate_data
+
+    @staticmethod
+    def to_json(certificate):
+        """
+        Serialize the Certificate object's locally-stored certificate data to a JSON representation
+        We use direct access here for specific keys in order to enforce their presence
+        """
+        # pylint: disable=no-member
+        certificate_data = certificate.get_certificate_data()
+        return {
+            "id": certificate_data['id'],
+            "name": certificate_data['name'],
+            "description": certificate_data['description'],
+            "version": CERTIFICATE_SCHEMA_VERSION,
+            "signatories": certificate_data['signatories']
+        }
+
+    @staticmethod
+    def from_json(course, value):
+        """
+        Deserialize from a JSON representation into a Certificate object.
+        'value' should be either a Certificate instance, or a valid JSON string
+        """
+
+        # Ensure the schema fieldset meets our expectations
+        for key in ("name", "description", "version"):
+            if key not in value:
+                raise CertificateValidationError(_("Certificate dict {0} missing value key '{1}'").format(value, key))
+
+        # Load up the Certificate data
+        certificate_data = CertificateManager.parse(value)
+        CertificateManager.validate(certificate_data)
+        certificate_data = CertificateManager.assign_id(course, certificate_data, certificate_data.get('id', None))
+        certificate = Certificate(course, certificate_data)
+
+        # Return a new Certificate object instance
+        return certificate
+
+    @staticmethod
+    def deserialize_certificate(course, json_string):
+        """
+        Returns a Certificate object instance given a valid course and JSON representation
+        """
+        return CertificateManager.from_json(course, json_string)
+
+    @staticmethod
+    def get_certificates(course):
+        """
+        Retrieve the certificates list from the provided course
+        """
+        # The top-level course field is 'certificates'
+        course_certificates_field = getattr(course, 'certificates')
+        # Inside of this top-level field are various things, including the actual 'certificates' list
+        certificates = course_certificates_field.get('certificates', [])
+        return certificates
+
+    @staticmethod
+    def remove_certificate(request, store, course, certificate_id):
+        """
+        Remove certificate from the course
+        """
+        for index, cert in enumerate(course.certificates['certificates']):
+            if int(cert['id']) == int(certificate_id):
+                course.certificates['certificates'].pop(index)
+                store.update_item(course, request.user.id)
+                break
+        return JsonResponse(status=204)
```

Comment:
None of the other methods (except `remove_signatory` below) return a `JsonResponse` - Seems like the calling view should be the thing to return the `JsonResponse`